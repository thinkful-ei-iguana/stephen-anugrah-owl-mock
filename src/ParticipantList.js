/* eslint-disable no-unused-expressions */
import React from 'react';
import Participant from './Participant';



const participantList = (props) => {

    return (
    <section className="participantList">
        {props.participants.map((participant) => {
            <Participant 
            key={participant.id}
            name={participant.name}
            avatar={participant.avatar}
            onStage={participant.onStage}
            inSession={participant.inSession}
            />
        })}
    </section>
    );
};

export default participantList;