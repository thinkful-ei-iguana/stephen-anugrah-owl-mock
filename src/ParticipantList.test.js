import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ParticipantList from './ParticipantList.js';
import Participant from './Participant';

const testArr= [{
    id: 1,
    name: 'Koren Templeton',
    avatar:
        'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
    inSession: true,
    onStage: true
},
{
    id: 2,
    name: 'Caty Flucker',
    avatar:
        'https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1',
    inSession: true,
    onStage: false
}];



describe('<ParticipantList />', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');


        ReactDOM.render(<ParticipantList 
           participants= {testArr}
        />, div);
    
        ReactDOM.unmountComponentAtNode(div);
    })   
})

describe('<ParticipantList />', () => {
    it('renders this UI as expected', () => {
        const list = renderer.create(<ParticipantList 
            participants= {testArr}
            />).toJSON();

        expect(list).toMatchSnapshot();

    });    
});