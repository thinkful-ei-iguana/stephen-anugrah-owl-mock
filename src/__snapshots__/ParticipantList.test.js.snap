// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ParticipantList /> renders this UI as expected 1`] = `
<section
  className="participantList"
/>
`;
